File: backend/src/main/java/cz/forgottenempire/servermanager/steamauth/SteamCmdSteamAuthVerifier.java
package cz.forgottenempire.servermanager.steamauth;

import cz.forgottenempire.servermanager.steamauth.AuthVerificationResult.AuthStatus;
import cz.forgottenempire.servermanager.steamauth.AuthVerificationResult.AuthType;
import cz.forgottenempire.servermanager.steamcmd.SteamCmdAuthService;
import cz.forgottenempire.servermanager.api.model.SteamAuthDto;
import lombok.extern.slf4j.Slf4j;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

@Service
@Slf4j
class SteamCmdSteamAuthVerifier implements SteamAuthVerifier {

    private final SteamCmdAuthService steamCmdAuthService;

    @Autowired
    SteamCmdSteamAuthVerifier(SteamCmdAuthService steamCmdAuthService) {
        this.steamCmdAuthService = steamCmdAuthService;
    }

    @Override
    public AuthVerificationResult verifyCredentials(SteamAuthDto authDto) {
        SteamAuth auth = new SteamAuth();
        auth.setUsername(authDto.getUsername());
        auth.setPassword(authDto.getPassword());
        auth.setSteamGuardToken(authDto.getSteamGuardToken());

        try {
            return steamCmdAuthService.verifyCredentials(auth);
        } catch (Exception e) {
            log.error("Error during credential verification", e);
            return AuthVerificationResult.builder()
                    .status(AuthStatus.ERROR)
                    .authType(AuthType.UNKNOWN)
                    .message("Verification failed: " + e.getMessage())
                    .build();
        }
    }
}

File: backend/src/main/kotlin/cz/forgottenempire/servermanager/steamauth/SteamCmdSteamAuthVerifier.kt
package cz.forgottenempire.servermanager.steamauth

import cz.forgottenempire.servermanager.api.model.SteamAuthDto
import cz.forgottenempire.servermanager.steamauth.AuthVerificationResult.AuthStatus
import cz.forgottenempire.servermanager.steamauth.AuthVerificationResult.AuthType
import cz.forgottenempire.servermanager.steamcmd.SteamCmdAuthService
import org.slf4j.LoggerFactory
import org.springframework.stereotype.Service

@Service
class SteamCmdSteamAuthVerifier(
    private val steamCmdAuthService: SteamCmdAuthService
) : SteamAuthVerifier {

    override fun verifyCredentials(authDto: SteamAuthDto): AuthVerificationResult {
        val auth = SteamAuth(
            username = authDto.username,
            password = authDto.password,
            steamGuardToken = authDto.steamGuardToken
        )
        return try {
            steamCmdAuthService.verifyCredentials(auth)
        } catch (e: Exception) {
            log.error("Error during credential verification", e)
            AuthVerificationResult.builder()
                .status(AuthStatus.ERROR)
                .authType(AuthType.UNKNOWN)
                .message("Verification failed: ${e.message}")
                .build()
        }
    }

    companion object {
        private val log = LoggerFactory.getLogger(SteamCmdSteamAuthVerifier::class.java)
    }
}
